### PR TITLE
added responsive Buttons to view the components

### PR DESCRIPTION
**Pull request by barczewski-espirit**
Here the possibility was implemented to view the components in their responsive representation. 
For this purpose the components are displayed one by one on a page which can be found under "/raw-component/{:component-name}". This page is integrated on the detail page of the component via iFrame. To change the view, buttons have been added which influence the width of the displayed iFrame.

### CODE REVIEW

File: src/documentation/index.tsx [hunk 1]
```diff
@@ -113,11 +113,12 @@ const routes = [
   name: "Documentation",
 })
 class Documentation extends BaseComponent {
-  showSidebar = false;
+  showSidebar = true;
```
Comment (vlapl) [inline src/documentation/index.tsx:116]: What is showSidebar used for?

File: tailwind.config.js [hunk 2]
```diff
@@ -349,6 +346,7 @@ module.exports = {
       "6xl": "72rem",
       full: "100%",
       ...breakpoints(theme("screens")),
+      "screen-xs": "600px",
     }),
```
Comment (vlapl) [inline tailwind.config.js:350]: I think 400px is better here (see my commit).

File: src/documentation/index.tsx (hunk 1 above)
Comment (barczewski-espirit) [reply src/documentation/index.tsx:116]: I think nothing at all. To be honest i don't know why i changed it. 
I'll remove it completely.

File: src/components/internal/Documentation/ComponentIFrame/ComponentIFrame.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  filename: string;
+  dark?: boolean;
+}
+
+@Component({
+  name: "ComponentIFrame",
+})
+class ComponentIFrame extends BaseComponent<ComponentIFrameProps> {
+  @Prop({ required: true }) filename!: ComponentIFrameProps["filename"];
+  @Prop() dark!: ComponentIFrameProps["dark"];
```
Comment (tgolbs) [inline src/components/internal/Documentation/ComponentIFrame/ComponentIFrame.tsx:16]: Please add `default: false` to the prop definition.

File: src/components/internal/Documentation/ComponentIFrame/ComponentIFrame.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+  render() {
+    return (
+      <div class="bg-gray-400">
+        <div
+          class={`flex space-x-4 justify-center ${
+            this.dark ? "bg-black" : "bg-white"
+          } text-xl`}
+        >
+          <ResponsiveButton
```
Comment (tgolbs) [inline src/components/internal/Documentation/ComponentIFrame/ComponentIFrame.tsx:44]: Please use a for each loop and add your classes to an array.

Comment (tgolbs) [reply src/components/internal/Documentation/ComponentIFrame/ComponentIFrame.tsx:44]:
```
["max-w-screen-xs", "max-w-screen-sm"].forEach((width, index) => {
    return (
        <ResponsiveButton
            dark={this.dark}
            pressed={this.pressedButtonIndex === index}
            handleOnClick={() => {
              this.pressedButtonIndex = index;
              this.iframeWidth = width;
            }}
          >
            <i class="fas fa-tablet-alt" />
          </ResponsiveButton>
    )
})
```

**Review by tgolbs — CHANGES_REQUESTED** [review]
Great job. Please fix the last small issues and we can integrate your work.

**Pull request closed without merging**